webapp/src/js/services/enketo.js

```diff
@@ -432,7 +432,8 @@ angular.module('inboxServices').service('Enketo',
 
       var attach = function(elem, file, type, alreadyEncoded, xpath) {
         xpath = xpath || xpathPath(elem);
-        var filename = 'user-file' + xpath;
+        // replace instance root element node name with form internal ID
+        var filename = xpath.replace(/^\/[^\/]+/, 'user-file' + '/' + doc.form);
```

Comment:
Almost all the time this is going to be a null operation as the root element is almost always the same as the form internal ID. Does it make sense to wrap this in a check to see if xpath already starts with doc.form?

Reply:
Sure.